### PR TITLE
hipfft: update @master dependency wrt rocfft

### PR DESCRIPTION
**Pull request by balay**
Without this fix:
```
$ ./bin/spack spec ginkgo+rocm amdgpu_target=gfx90a ^hip@5.7.1 |egrep '(hip|hipfft)@'
 -       ^hip@5.7.1
 -       ^hip@5.7.1%gcc@11.4.0~asan~cuda~ipo+rocm build_system=cmake build_type=Release generator=make patches=5bb9b0e,7668b2a,aee7249,b589a02,c2ee21c arch=linux-ubuntu22.04-x86_64
 -       ^hipfft@master%gcc@11.4.0~asan~cuda~ipo+rocm amdgpu_target=auto build_system=cmake build_type=Release generator=make arch=linux-ubuntu22.04-x86_64
```
With this fix:
```
$ ./bin/spack spec ginkgo+rocm amdgpu_target=gfx90a ^hip@5.7.1 |egrep '(hip|hipfft)@'
 -       ^hip@5.7.1
 -       ^hip@5.7.1%gcc@11.4.0~asan~cuda~ipo+rocm build_system=cmake build_type=Release generator=make patches=5bb9b0e,7668b2a,aee7249,b589a02,c2ee21c arch=linux-ubuntu22.04-x86_64
 -       ^hipfft@5.7.1%gcc@11.4.0~asan~cuda~ipo+rocm amdgpu_target=auto build_system=cmake build_type=Release generator=make arch=linux-ubuntu22.04-x86_64
```

### CODE REVIEW

**Comment by alalazo** [comment]
Can you explain how this solves the issue (i.e. which dependency constraints are involved in the selection)?

**Comment by balay** [comment]
> Can you explain how this solves the issue (i.e. which dependency constraints are involved in the selection)?

Not really. I don't understand why spack chooses `hipfft@master` for `hip@:5.7.1` [works fine for `hip@6.0.0:`] - and this issue occurred only after  #45701 was merged.

My view with this fix is - "some hip/rocm pkgs have `@master` dependency listed - and others don't have it. So add more to this `@master` dependency list - so this might be a desired change either way".

Will defer the proper diagnosis and fix of the primary issue to hip maintainers.

**Comment by balay** [comment]
Additional note: with the current develop snapshot - I can use the following workaround:
```
svcpetsc@petsc-gpu-02:/scratch/svcpetsc/spack.a$ ./bin/spack spec ginkgo+rocm amdgpu_target=gfx90a ^hip@5.7.1 ^hipfft@5.7.1 |egrep '(hip|hipfft)@'
 -       ^hip@5.7.1
 -       ^hipfft@5.7.1
 -       ^hip@5.7.1%gcc@11.4.0~asan~cuda~ipo+rocm build_system=cmake build_type=Release generator=make patches=5bb9b0e,7668b2a,aee7249,b589a02,c2ee21c arch=linux-ubuntu22.04-x86_64
 -       ^hipfft@5.7.1%gcc@11.4.0~asan~cuda~ipo+rocm amdgpu_target=auto build_system=cmake build_type=Release generator=make arch=linux-ubuntu22.04-x86_64
```

**Review by afzpatel — APPROVED** [review]

**Review by renjithravindrankannath — APPROVED** [review]
LGTM

**Review by tldahlgren — APPROVED** [review]
Confirmed the branch exists.

**Pull request merged**
